tensorflow/lite/micro/tools/make/ext_libs/cmsis_nn.inc

```diff
@@ -56,5 +72,28 @@ THIRD_PARTY_CC_HDRS += \
 INCLUDES += \
   -I$(CMSIS_PATH) \
   -I$(CMSIS_NN_PATH) \
+  -I$(CMSIS_DSP_PATH) \
   -I$(CMSIS_PATH)/CMSIS/Core/Include \
-  -I$(CMSIS_NN_PATH)/Include
+  -I$(CMSIS_NN_PATH)/Include \
+  -I$(CMSIS_DSP_PATH)/Include \
+  -I$(CMSIS_DSP_PATH)/PrivateInclude
+
+MIN_RFFT_Q15_LEN := 32
+MAX_RFFT_Q15_LEN := 8192
+
+MIN_RFFT_Q31_LEN := 32
+MAX_RFFT_Q31_LEN := 8192
+
+MIN_RFFT_FLOAT_LEN := 32
+MAX_RFFT_FLOAT_LEN := 4096
+
+CMSIS_NN_FLAGS := \
+  -DMIN_RFFT_Q15_LEN=$(MIN_RFFT_Q15_LEN) \
+  -DMAX_RFFT_Q15_LEN=$(MAX_RFFT_Q15_LEN) \
+  -DMIN_RFFT_Q31_LEN=$(MIN_RFFT_Q31_LEN) \
+  -DMAX_RFFT_Q31_LEN=$(MAX_RFFT_Q31_LEN) \
+  -DMIN_RFFT_FLOAT_LEN=$(MIN_RFFT_FLOAT_LEN) \
+  -DMAX_RFFT_FLOAT_LEN=$(MAX_RFFT_FLOAT_LEN)
+
+CCFLAGS += $(CMSIS_NN_FLAGS)
+CXXFLAGS += $(CMSIS_NN_FLAGS)
```

Comment:
I'm lookng at https://github.com/ARM-software/CMSIS-DSP to see what the different compiler flags are.

Do you have any profiling information that tells us the impact of `-Ofast`, `-O2`, `-DARM_MATH_LOOPUNROLL` etc. also, is it worth disabling f16 (`-DDISABLEFLOAT16`)?

Reply:
I haven't experimented with flags.

Reply:
Is there a way for us to specify flags that will affect cmsis-dsp sources but not other sources?

Reply:
We have a way for specifying the optimization level for third_party kernel sources:
https://github.com/tensorflow/tflite-micro/blob/main/tensorflow/lite/micro/tools/make/Makefile#L706

It might be good to extend that with a new flag for `THIRD_PARTY_KERNEL_CCFLAGS`.

Reply:
I'm hesitant to define -Ofast, -O2 even for third party kernels. If possible, I'd like to land this PR with the default flags (-O2).

Reply:
These are the recommended options to use: https://github.com/ARM-software/CMSIS-DSP#options-to-use
So for performance:
-Ofast 
-DLOOPUNROLL=ON
If you go with the default for kernels -O2 there will probably not be a huge performance diff. 
If using GCC and Helium (e.g. Arm(R) Cortex(R)-M55) this will be faster but may require some additional tweaks to get integrated: https://github.com/ARM-software/EndpointAI/tree/master/Kernels/ARM-Optimized/DSP/Source
About F16, it can be useful for Helium: https://github.com/ARM-software/CMSIS-DSP#half-float-support

Reply:
Thanks @mansnils. @advaitjain, how should we proceed? Regarding -Ofast, since there's not much difference from -O2, I'd rather not s/-O2/-Ofast for fear of breaking/bloating the rest of the code. Ideally, we should have flags that are specific to each external library, but since we don't, I'd rather punt. 
Regarding f16 and loop unroll, we can control them with make variables.